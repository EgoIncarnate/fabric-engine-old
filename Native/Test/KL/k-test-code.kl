function double( io Scalar input, io Scalar output )
{
  output = 2 * input;
}

function entry()
{
  var Scalar input = 4.0, output;
  double( input, output );
  report output;
}
