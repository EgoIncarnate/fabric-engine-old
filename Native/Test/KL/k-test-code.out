GlobalList
  Function( 'double' ('double'), void )
    ParamList
      Param( 'input', Scalar, l-value )
      Param( 'output', Scalar, l-value )
    CompoundStatement
      StatementList
        ExprStatement
          AssignOp(ASSIGN_OP)
            Var('output')
            BinOp( MUL )
              ConstSize( 2 )
              Var('input')
  Function( 'entry' ('entry'), void )
    ParamList
    CompoundStatement
      StatementList
        StatementList
          AssignedVarDecl( VarDecl( 'input', Scalar ), ConstScalar( 4 ) )
            ConstScalar( 4 )
          VarDecl( 'output', Scalar )
        ExprStatement
          Call('double')
            ArgList
              Var('input')
              Var('output')
        Report
          Var('output')
8
